numpy/lib/tests/test_arraypad.py

```diff
@@ -1016,6 +1021,8 @@ def test_check_negative_pad_width(self):
     def test_check_empty_array(self):
         assert_raises(ValueError, pad, [], 4, mode='reflect')
         assert_raises(ValueError, pad, np.ndarray(0), 4, mode='reflect')
+        assert_raises(ValueError, pad, np.zeros((0, 3)), ((1,), (0,)),
+                      mode='reflect')
```

Comment:
Can you add a test of the allowed case too?

Reply:
`pad(np.zeros((0, 3)), ((0,), (1,)), mode='reflect')` would do, I think?

Reply:
Might want to assert the new shape too.

Reply:
I do not think it should go into `TestValueError` test suite, so I have added this exact case to `TestReflect` .